yaml_edit/lib/src/editor.dart

```diff
@@ -628,11 +602,11 @@ class YamlEditor {
         throw PathError(path, subPath, tree);
       }
 
-      return wrapAsYamlNode([...tree.nodes]..[keyOrIndex] = _deepModify(
+      return wrapAsYamlNode([...tree.nodes]..[keyOrIndex as int] = _deepModify(
```

Comment:
See here we have:
```
if (!isValidIndex(keyOrIndex, tree.length)) {
  throw PathError(path, subPath, tree);
}
```
So this cast is known to be safe, and we will throw the right error :D

Reply:
Yup. Also please see my comment on this behavior on `update()`.